### PR TITLE
refactor(touchspin): convert touchspin to PF4

### LINKED ISSUE #148: Component: Touchspin to PF4
## Acceptance Criteria

- Convert Touchspin instances to PF4 component

## Additional context

...
Issue for tracking refactor to match PF4 styling/implementation of the `NumberInput` component.

#### Related issue
[DISCOVERY-153](https://issues.redhat.com/browse/DISCOVERY-153)
#104

### PR DESCRIPTION
**Pull request by jenny-s51**
Closes #148 

### Before
Screen Shot 2022-08-26 at 12 13 56 PM

### After
Screen Shot 2022-08-26 at 12 12 56 PM

#### Related issues
[DISCOVERY-153](https://issues.redhat.com/browse/DISCOVERY-153)
#104

### CODE REVIEW

File: src/components/createScanDialog/createScanDialog.js [hunk 1]
```diff
@@ -185,7 +185,7 @@ class CreateScanDialog extends React.Component {
     );
   }
 
-  renderConcurrentScans({ values, handleOnEvent }) {
+  renderConcurrentScans({ values }) {
```
Comment (cdcabrera) [inline src/components/createScanDialog/createScanDialog.js:188]: required:
lets restore this, we think it's the cause of not submitting values... probably need to restore the "mockEvent" pieces too

**Review by cdcabrera — CHANGES_REQUESTED** [review]
And lets restore the naming from numberInput to touchspin... that'll help us with history/tracking.

**Review by bclarhk — APPROVED** [review]
silly number spinners. LGTM!

File: src/components/touchspin/touchspin.js [hunk 2]
```diff
@@ -59,46 +72,39 @@ class TouchSpin extends React.Component {
...
-  onUpdateValue = event => {
-    const { maxValue, minValue } = this.state;
+  onUpdateValue = (_updatedValue, event) => {
     const { name, onChange } = this.props;
-    const { value } = event.target;
+    const { maxValue, minValue } = this.state;
 
-    let parsedValue = Number.parseInt(value, 10);
-    parsedValue = Number.isNaN(parsedValue) ? minValue : parsedValue;
+    let parsedValue = Number(event.target.value);
```
Comment (cdcabrera) [inline src/components/touchspin/touchspin.js:95]: required:
`Number to Number.parseInt(yourstringnumberhere, 10)`

**Review by cdcabrera — CHANGES_REQUESTED** [review]

File: src/components/touchspin/touchspin.js [hunk 3]
```diff
@@ -59,46 +72,39 @@ class TouchSpin extends React.Component {
...
 
-  onUpdateValue = event => {
-    const { maxValue, minValue } = this.state;
+  onUpdateValue = (_updatedValue, event) => {
     const { name, onChange } = this.props;
-    const { value } = event.target;
+    const { maxValue, minValue } = this.state;
 
-    let parsedValue = Number.parseInt(value, 10);
-    parsedValue = Number.isNaN(parsedValue) ? minValue : parsedValue;
```
Comment (cdcabrera) [inline src/components/touchspin/touchspin.js:90]: required:
going back through, this NaN check needs to be restored too, and possibly the rest of lines `84` through `90`

reason:
- Input fields regardless of whether they're declared "number" still return a string. the joys of HTML/JS interaction.
- this also covers the scenario when someone enters nothing but word, one of the reasons we call `parseInt`

**Review by cdcabrera — CHANGES_REQUESTED** [review]

**Review by cdcabrera — APPROVED** [review]
lgtm! much appreciated on the patience @jenny-s51

**Pull request merged**